restaurants/src/app/components/restaurants/restaurants.component.html

```diff
@@ -14,23 +14,18 @@ <h2>Restaurants</h2>
     <label for="rank">Rank:</label>
     <input id="rank" [(ngModel)]="rank" placeholder="rank">
 
-    <label for="id">ID:</label>
-    <input id="id" [(ngModel)] = "id" placeholder="id">
-
-    <button type="button" (click)="add(id, name, rank)">Add Restaurant</button>
+    <button type="button" (click)="add(name, rank)">Add Restaurant</button>
   </div>
 
   <div *ngIf="selectedRestaurant">
-    <h2>{{selectedRestaurant.name | uppercase}} Details</h2>
+    <h2>{{restaurants[selectedRestaurant].name | uppercase}} Details</h2>
     <div>
       <label for="restaurant-name">Restaurant name: </label>
-      <input id="restaurant-name" [(ngModel)]="selectedRestaurant.name" placeholder="name">
-      <div>Rank: {{selectedRestaurant.rank}}</div>
-      <label for="restaurant-id">ID: </label>
-      <input id="restaurant-id" [(ngModel)]="selectedRestaurant.id" placeholder="id">
+      <input id="restaurant-name" [(ngModel)]="newRestaurant.name" placeholder="name">
+      <div>Rank: {{newRestaurant.rank}}</div>
     </div>
   </div>
 
   <div *ngIf="selectedRestaurant">
-    <button type="button" (click)="delete(selectedRestaurant.id)">Delete Restaurant</button>
-  </div>
+    <button type="button" (click)="delete(selectedRestaurant)">Delete Restaurant</button>
```

Comment:
Since you can only ever delete the selected restaurant, do you need to pass `selectedRestaurant` in as a parameter to the method? Seems like you don't need a parameter for the `delete()` function, and instead can just use `this.selectedRestaurant` inside that method.

Reply:
i don't think i can because i need the key?